### ISSUE TITLE
Can't test for None in array (Trac #1010)

### ISSUE THREAD

**Issue by thouis** [numpy/numpy#1608]
_Original ticket http://projects.scipy.org/numpy/ticket/1010 on 2009-02-19 by trac user anjiro, assigned to unknown._

Testing for `None` in an array always returns `False`:

```
>>> import numpy as np
>>> 5 in np.array([None, 5, 23])
True
>>> None in np.array([None, 5, 23])
False
```

**Comment by Nodd** [comment]
It works now:
```python
>>> None in np.array([None, 5, 23])
True
>>> None in np.array([2, 5, 23])
False
```

**Comment by jakirkham** [comment]
Sounds like a good time to add a test to lock in the good expected behavior. 😉

**Comment by eric-wieser** [comment]
This became a warning in #3514, and was fixed in #8297

**Comment by eric-wieser** [comment]
Given we have this test, I'd say we're good:

https://github.com/charris/numpy/blob/maintenance/1.14.x/numpy/core/tests/test_numeric.py#L1269-L1277

Feel free to add one for `in` alongside it if you want to though, @Nodd
